avaframe/ana3AIMEC/ana3AIMEC.py

```diff
@@ -733,23 +785,61 @@ def analyzeArea(rasterTransfo, resAnalysis, pLim, newRasters, cfgPath, cfgFlags)
 def readWrite(fname_ent):
     """
     Read mass balance files to get mass properties of the simulation
-    (total mass, entrained mass...)
+    (total mass, entrained mass...). Checks for mass conservation
+    input:
+            -fname_ent: list of mass balance files
+    output:
+            -relMass: release mass
+            -entMass: entrained mass
+            -growthIndex
+            -growthGrad
     """
     #    load data
     #    time, total mass, entrained mass
     massTime = np.loadtxt(fname_ent, delimiter=',', skiprows=1)
     timeResults = [massTime[0, 0], massTime[-1, 0]]
     totMassResults = [massTime[0, 1], massTime[-1, 1]]
-    entMassResults = [massTime[0, 2], massTime[-1, 2]]
     relMass = totMassResults[0]
-    entMass = entMassResults[1]
+    entMass = np.sum(massTime[:, 2])
+    finalMass = totMassResults[1]
+    # check mass balance
+    log.info('Mass difference between first and last time step in sim %s is: %.1f kg' %
+             (int(os.path.splitext(os.path.basename(fname_ent))[0]), totMassResults[1]- relMass))
+    log.info('Entrained mass in sim %s is: %.1f kg' % (int(os.path.splitext(os.path.basename(fname_ent))[0]), entMass))
+    if (totMassResults[1]- relMass)==0:
+        diff = np.abs((totMassResults[1]- relMass) - entMass)
+        if diff>0:
+            log.warning('Conservation of mass is not satisfied')
+            log.warning('Mass difference and entrained mass differ from %.4f kg' % (diff))
```

Comment:
I would suggest to write: "Total mass change and total entrained mass differ about ..."